expression/builtin_time_vec.go

```diff
@@ -1116,11 +1116,52 @@ func (b *builtinToDaysSig) vecEvalInt(input *chunk.Chunk, result *chunk.Column)
 }
 
 func (b *builtinDateFormatSig) vectorized() bool {
-	return false
+	return true
 }
 
+// vecEvalString evals a builtinDateFormatSig.
+// See https://dev.mysql.com/doc/refman/5.7/en/date-and-time-functions.html#function_date-format
 func (b *builtinDateFormatSig) vecEvalString(input *chunk.Chunk, result *chunk.Column) error {
-	return errors.Errorf("not implemented")
+	n := input.NumRows()
+	buf, err := b.bufAllocator.get(types.ETDatetime, n)
+	if err != nil {
+		return err
+	}
+	defer b.bufAllocator.put(buf)
+	if err := b.args[0].VecEvalTime(b.ctx, input, buf); err != nil {
+		return err
+	}
+
+	buf0, err := b.bufAllocator.get(types.ETString, n)
+	if err != nil {
+		return err
+	}
+	defer b.bufAllocator.put(buf0)
+	if err := b.args[1].VecEvalString(b.ctx, input, buf0); err != nil {
+		return err
+	}
+
+	result.ReserveString(n)
+	ds := buf.Times()
+	for i := 0; i < n; i++ {
+		if buf.IsNull(i) || buf0.IsNull(i) {
+			result.AppendNull()
+			continue
+		}
+		str, err := ds[i].DateFormat(buf0.GetString(i))
+		if err != nil {
+			return err
+		}
```

Comment:
We should solve the case when the ```str = "" ```, like [here](https://github.com/pingcap/tidb/blob/master/expression/builtin_time.go#L798-L802). We can add
```
    // MySQL compatibility, #11203
	// If format mask is 0 then return 0 without warnings
	if str == "0" {
		result.AppendString("0")
        continue
	}
```